app/controllers/publishers/ats_api/v1/create_vacancy_service.rb

```diff
@@ -0,0 +1,77 @@
+module Publishers
+  module AtsApi
+    module V1
+      class CreateVacancyService
+        InvalidOrganisationError = Class.new(StandardError)
+
+        class << self
+          def call(params)
+            vacancy = Vacancy.new(permitted_params(params))
+
+            if (conflict = conflict_vacancy(vacancy))
+              return conflict_response(conflict)
+            end
+
+            if vacancy.save
+              success_response(vacancy)
+            else
+              validation_error_response(vacancy)
+            end
+          end
+
+          private
+
+          def permitted_params(params)
```

Comment:
I find the (almost) duplication of this between create and update a little confusing. This one throws InvalidOrganisationError, the other throws RecordNotFound.